nixos/modules/services/web-servers/bluemap.nix

```diff
@@ -267,6 +265,12 @@ in {
             You can achieve this through setting `services.bluemap.eula = true`
           '';
         }
+        { assertion = cfg.enableNginx -> cfg.host != null;
+          message = ''
+            Nginx is enabled to serve the bluemap webapp, but you have not assigned a host
+            to use. You can do this by setting `services.bluemap.host = "hostname"`.
+          '';
+        }
```

Comment:
I think unfortunately since the option now doesn't have a default it won't be null.

Instead it will  just not be present in the configuration. That means you can't have an assertion on this being null. You have to just trust the module system to complain that it is undefined (with a slightly worse error message), rather than having this manual assertion

```suggestion
```

Reply:
Right you are. I tested this out, and the error that results when your suggestion is applied is this:

```
error:
       … while calling the 'head' builtin
         at /nix/store/9gsbvj93aqfffkslk9fsrw3af78s1kz7-source/lib/attrsets.nix:1575:11:
         1574|         || pred here (elemAt values 1) (head values) then
         1575|           head values
             |           ^
         1576|         else

       … while evaluating the attribute 'value'
         at /nix/store/9gsbvj93aqfffkslk9fsrw3af78s1kz7-source/lib/modules.nix:809:9:
          808|     in warnDeprecation opt //
          809|       { value = builtins.addErrorContext "while evaluating the option `${showOption loc}':" value;
             |         ^
          810|         inherit (res.defsFinal') highestPrio;

       … while evaluating the option `system.build.toplevel':

       … while evaluating definitions from `/nix/store/9gsbvj93aqfffkslk9fsrw3af78s1kz7-source/nixos/modules/system/activation/top-level.nix':

       … while evaluating the option `assertions':

       … while evaluating definitions from `/nix/store/9gsbvj93aqfffkslk9fsrw3af78s1kz7-source/nixos/modules/system/boot/systemd.nix':

       … while evaluating the option `systemd.services':

       … while evaluating definitions from `/nix/store/9gsbvj93aqfffkslk9fsrw3af78s1kz7-source/nixos/modules/security/acme':

       … while evaluating the option `security.acme.certs':

       … while evaluating definitions from `/nix/store/9gsbvj93aqfffkslk9fsrw3af78s1kz7-source/nixos/modules/services/web-servers/nginx/default.nix':

       … while evaluating the option `services.nginx.virtualHosts':

       … while evaluating definitions from `/nix/store/9gsbvj93aqfffkslk9fsrw3af78s1kz7-source/nixos/modules/services/web-servers/bluemap.nix':

       … while evaluating the option `services.bluemap.host':

       (stack trace truncated; use '--show-trace' to show the full, detailed trace)

       error: The option `services.bluemap.host' is used but not defined.
```

That's definitely good enough.